File: backend/src/main/java/com/artifax_solutions/website/model/SocialSection.java
package com.artifax_solutions.website.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class SocialSection {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String subTitle;
    private String buttonText;
    private String description;
    @Column(columnDefinition = "TEXT")
    private String buttonIconSvg;
    private String profileImageUrl;
    private String profileImageAltText;
    @OneToMany(mappedBy = "socialSection")
    private List<SocialPost> postList;
    @OneToMany(mappedBy = "socialSection")
    private List<SocialMediaLink> socialMediaLinks;
}

File: frontend/src/interface/SocialSection.ts
export interface SocialSection {
  id: number | undefined | null;
  title: string | undefined | null;
  subTitle: string | undefined | null;
  description: string | undefined | null;
  profileImageUrl: string | undefined | null;
  profileImageAltText: string | undefined | null;
  postList: SocialPost[] | undefined | null;
  buttonIconSvg: string | undefined | null;
  buttonText: string | undefined | null;
  socialMediaLinks: SocialMediaLink[] | undefined | null;
}

export interface SocialPost {
  id: number | undefined | null;
  imageUrl: string | undefined | null;
  altText: string | undefined | null;
}

export interface SocialMediaLink {
  id: number | undefined | null;
  url: string | undefined | null;
  icon: string | undefined | null;
  platform: string | undefined | null;
}
